src/main/java/io/th0rgal/oraxen/mechanics/provided/gameplay/furniture/FurnitureMechanic.java

```diff
@@ -258,7 +258,13 @@ public boolean removeSolid(World world, BlockLocation rootBlockLocation, float o
                     seat.remove();
                 }
             }
-            location.getBlock().setType(Material.AIR);
+            final Block block = location.getBlock();
+            final PersistentDataContainer blockData = new CustomBlockData(block, OraxenPlugin.get());
+            blockData.remove(FURNITURE_KEY);
+            blockData.remove(SEAT_KEY);
+            blockData.remove(ROOT_KEY);
+            blockData.remove(ORIENTATION_KEY);
+            block.setType(Material.AIR);
```

Comment:
This might be an issue with farmblock and stuff aswell.
From talking with the dev of CustomBlockData 2.0.0 was meant to automatically remove this.
Reading over it seems we forgot to register this.

Could you try this and see if farmblock data is also left over?
And also register the listener for it in `OraxenPLugin`, should just be `CustomBlockData.registerListener(OraxenPlugin.get());`.
I think this might handle most of these for us